### PR TITLE
plugin fixes to work with firebase core

### PR DESCRIPTION
**Pull request by helenaford**

### CODE REVIEW

File: packages/firebase_remote_config/example/pubspec.yaml [hunk 1]
```diff
@@ -22,6 +22,6 @@ dev_dependencies:
   e2e: ^0.2.0
 
 environment:
-  sdk: ">=2.0.0-dev.28.0 <3.0.0"
-  flutter: ">=1.9.1+hotfix.2 <2.0.0"
+  sdk: ">=2.7.0 <3.0.0"
```
Comment (Salakar) [inline packages/firebase_remote_config/example/pubspec.yaml:25]: Could we change these to be 2.1.0 rather than 2.7.0, thanks 😊

Comment (helenaford) [reply packages/firebase_remote_config/example/pubspec.yaml:25]: yeah sure. admob used 2.7.0. but can change that too

File: packages/firebase_core/firebase_core/lib/src/firebase_app.dart [hunk 2]
```diff
@@ -43,13 +43,13 @@ class FirebaseApp {
...
   }
 
-  @Deprecated("Deprecated in favor of FirebaseCore.instance.initializeApp()")
-  static Future<FirebaseApp> configure({
-    @required String name,
-    @required FirebaseOptions options,
-  }) {
-    return FirebaseCore.instance.initializeApp(name: name, options: options);
-  }
+  // @Deprecated("Deprecated in favor of FirebaseCore.instance.initializeApp()")
```
Comment (helenaford) [inline packages/firebase_core/firebase_core/lib/src/firebase_app.dart:46]: should this be deleted @Salakar?

Comment (Salakar) [reply packages/firebase_core/firebase_core/lib/src/firebase_app.dart:46]: Keeps this one, we commented it out just flag any existing uses in the codebase that would need changing. But for user-land code we  should keep the deprecation.

Can remove the constructor one above (`FirebaseApp({@required String name})`) though - no point keeping that as a deprecation when the options are broken

Thanks

File: packages/firebase_remote_config/example/pubspec.yaml (hunk 1 above)
Comment (Salakar) [reply packages/firebase_remote_config/example/pubspec.yaml]: aye, don't think the admob change on master was right 🙈

File: packages/cloud_functions/cloud_functions/test/test_common.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const kCollectionId = "test";
+const kDocumentId = "document";
+
+void initializeMethodChannel() {
+  // Install the Codec that is able to decode FieldValues.
+  MethodChannelCloudFunctions.channel = MethodChannel(
+    'plugins.flutter.io/cloud_firestore',
+    StandardMethodCodec(TestFunctionsMessageCodec()),
+  );
```
Comment (Salakar) [inline packages/cloud_functions/cloud_functions/test/test_common.dart:20]:
```suggestion
```

Not needed - guessing copied from Firestore

File: packages/cloud_functions/cloud_functions/test/test_common.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import 'package:cloud_functions_platform_interface/cloud_functions_platform_interface.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:firebase_core_platform_interface/firebase_core_platform_interface.dart';
+
+typedef MethodCallCallback = dynamic Function(MethodCall methodCall);
+
+const kCollectionId = "test";
+const kDocumentId = "document";
```
Comment (Salakar) [inline packages/cloud_functions/cloud_functions/test/test_common.dart:13]:
```suggestion
```

Not needed - guessing copied from Firestore

File: packages/cloud_functions/cloud_functions/test/test_common.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  });
+}
+
+void handleMethodCall(MethodCallCallback methodCallCallback) =>
+    MethodChannelCloudFunctions.channel.setMockMethodCallHandler((call) async {
+      expect(
+          call.arguments["app"], equals(FunctionsPlatform.instance.app.name));
+      expect(call.arguments["path"], equals("$kCollectionId/$kDocumentId"));
+      return await methodCallCallback(call);
+    });
```
Comment (Salakar) [inline packages/cloud_functions/cloud_functions/test/test_common.dart:55]:
```suggestion
```

Not needed - guessing copied from Firestore

**Comment by helenaford** [comment]
Outstanding

Analyze:
* core/core_platform_interface

Tests for the following packages are failing (see above):
 * cloud_firestore/cloud_firestore_web
 * firebase_auth/firebase_auth_web

Not failing but needs to be looked at
* cloud functions example tests (commented out).

**Pull request merged**